### PR TITLE
Fix webstuff when user css does not exist

### PR DESCRIPTION
**Pull request by enlite93**
As mentioned in https://github.com/gramps-project/gramps/commit/74ffc76af0d3be2463a8ef918a82a8e1df4d6126#commitcomment-21026601
```

Traceback (most recent call last):
  File "/home/travis/build/gramps-project/gramps/gramps/gen/plug/_manager.py", line 181, in reg_plugins
    results = mod.load_on_reg(dbstate, uistate, plugin)
  File "/home/travis/build/gramps-project/gramps/gramps/plugins/webstuff/webstuff.py", line 181, in load_on_reg
    list_files = os.listdir(USER_CSS)
FileNotFoundError: [Errno 2] No such file or directory: '/home/travis/.gramps/css'
```

### CODE REVIEW

File: gramps/plugins/webstuff/webstuff.py [hunk 1]
```diff
@@ -178,12 +178,15 @@ def load_on_reg(dbstate, uistate, plugin):
...
-                          None, [], [] ])
-    return CSS_FILES
+    try:
+        list_files = os.listdir(USER_CSS)
+        for cssfile in list_files:
+            CSS_FILES.append([cssfile, 1, cssfile.replace('.css', ''),
+                              os.path.join(USER_CSS,cssfile),
+                              None, [], [] ])
+        return CSS_FILES
+    except:
```
Comment (QuLogic) [inline gramps/plugins/webstuff/webstuff.py]: Better not to use a bare `except`.

Comment (enlite93) [inline gramps/plugins/webstuff/webstuff.py]: What do you suggest?

Comment (QuLogic) [inline gramps/plugins/webstuff/webstuff.py]: Well, you should catch only what you expect to see; `except FileNotFoundError:` or `except OSError:` or something like that.

Comment (sam-m888) [inline gramps/plugins/webstuff/webstuff.py]: I'd suggest you just use

`except FileNotFoundError:`

**Comment by sam-m888** [comment]
Worked for me. 👍

**Comment by Nick-Hall** [comment]
Is this only a problem for the unit tests?  If so, why don't we just create the directory in the Travis environment?

**Comment by daleathan** [comment]
> Is this only a problem for the unit tests? If so, why don't we just create the directory in the Travis environment?

@Nick-Hall You already do that on Travis-CI when testing the addons-source https://github.com/gramps-project/addons-source/blob/master/.travis.yml#L80 but not when testing Gramps source.  But really from my point of view the problem is that when I run the test locally I get the same error and in effect your say that I now have to remember to create that css directory before the test are run isn't that a little backward. Another benefit of adding this is that you can then remove that line from Travis-CI for the addons-sources.

**Comment by Nick-Hall** [comment]
@daleathan OK. I see the problem. Why not just add an `if` statement?

`if os.path.exists(USER_CSS):`

**Comment by prculley** [comment]
Regarding the Travis failure; I had to (painfully) debug an issue just like this a while back. It occurs when the order of Travis tests changes such that a test is run that needs plugins, but the test does not include the creation of user directories or is not preceded by tests with code that does this.

I fixed it at the time for tests that use the test_util 'run()' method. Perhaps the failing test uses some other method to operate the Gramps code. See https://github.com/gramps-project/gramps/pull/336/files, the changes for the test_util file.

**Comment by enlite93** [comment]
> Why not just add an if statement? 
> 
> `if os.path.exists(USER_CSS):`
> 

@Nick-Hall Ok, so instead of try/except I'll use an if statement 👍

**Comment by Nick-Hall** [comment]
Fixed in commit c352fa63783ee6ceae689d4ce0c6ad67e2aa88bc.

**Comment by enlite93** [comment]
Was it something I did? Sure my solution wasn't exactly like yours!

Either way errors gone, so thanks I guess?

**Pull request closed without merging**